ffpass/__init__.py

```diff
@@ -216,44 +358,82 @@ def exportLogins(key, jsonLogins):
     for row in jsonLogins["logins"]:
         if row.get("deleted"):
             continue
-        encUsername = row["encryptedUsername"]
-        encPassword = row["encryptedPassword"]
-        logins.append(
-            (
-                row["hostname"],
-                decodeLoginData(key, encUsername),
-                decodeLoginData(key, encPassword),
-            )
-        )
+        try:
+            user = decodeLoginData(key, row["encryptedUsername"])
+            pw = decodeLoginData(key, row["encryptedPassword"])
+            logins.append((row["hostname"], user, pw))
+        except Exception as e:
+            if logging.getLogger().isEnabledFor(logging.DEBUG):
+                logging.debug(f"Failed to decrypt {row.get('hostname')}: {e}")
+            continue
     return logins
 
 
-def lower_header(csv_file):
-    it = iter(csv_file)
-    yield next(it).lower()
-    yield from it
+def readCSV(csv_file):
 
+    reader = csv.reader(csv_file)
 
-def readCSV(csv_file):
     logins = []
-    reader = csv.DictReader(lower_header(csv_file))
-    for row in reader:
-        logins.append((rawURL(row["url"]), row["username"], row["password"]))
-    logging.info(f'read {len(logins)} logins')
+    first_row = None
+
+    # Loop through logins
+    for i, row in enumerate(reader):
+
+        logging.debug(f"row: {row}")
+
+        # Peek at the first line to detect if it is a header or normal row
+        if first_row is None:
```

Comment:
This is convoluted logic, which is perhaps best left to the user as a `--header` flag, without which input is assumed to be headerless (i.e., first row is already an `(url, username, password)` tuple/list?

Or default the other way if you prefer.

But this feels excessive, when I look at it with fresh eyes. Not tested on export to file, either, unsure if there is/was a test for that.